src/clients/FtpRelay.cc

```diff
@@ -567,7 +567,7 @@ Ftp::Relay::readReply()
     assert(serverState() == fssConnected ||
            serverState() == fssHandleUploadRequest);
 
-    if (100 <= ctrl.replycode && ctrl.replycode < 200)
+    if (Http::Is1xx(ctrl.replycode))
```

Comment:
It seems rather odd to checking HTTP protocol status on FTP protocol messages. Perhapse just de-duplicate the checks into an FtpRelay method rather than depending on the HTTP status checker.

Reply:
The notion of 1xx responses exists in both protocols. I suggest adding Ftp::Is1xx() to src/ftp/Elements.h. The new inlined function will duplicate Http::Is1xx() implementation, but that is OK IMO: We could convert both functions to call some common range checking function, but I do not think that complexity is better than duplicating this simple range checking code twice.

Reply:
> I suggest adding Ftp::Is1xx() to src/ftp/Elements.h.

Done.